src/GraphQL.Conventions/Types/Resolution/ICanChangeReflectorSettings.cs

```diff
@@ -0,0 +1,65 @@
+﻿using GraphQL.Conventions.Types.Resolution;
+using System.Linq;
+
+namespace GraphQL.Conventions
+{
+    public interface ICanChangeReflectorSettings
+    { }
+
+    public static class ReflectorSettingsExtensions
+    {
+        static object _lockKey = new object();
+
+        public static T IgnoreTypesFromNamespacesStartingWith<T>(this T source, params string[] namespacesToIgnore) where T : ICanChangeReflectorSettings
```

Comment:
Instead of using an extension method on `ICanChangeReflectorSettings`, can you please remove the interface and provide this functionality through the `ITypeResolver` interface instead? See how this is done with `RegisterScalarType<TType>(string typeName)`, and how it's wired up with `GraphQLEngine`